cogs/mediaconverter.py

```diff
@@ -17,79 +16,74 @@ async def on_message(self, message: Message):
         if message.author.bot:
             return
 
-        link_type, link_url = self.extract_link(message.content)
-        if link_type:
-            description = ""
-            if link_type == "tiktok":
-                embedded_video = self.embed_tiktok(link_url)
-                description = f"[TikTok Link]({embedded_video})"
+        link_type, link_url = MediaLinkFormatter.detect_link(message.content)
 
-                await message.edit(
-                    suppress_embeds=True
-                )  # Removes previous embed from context message
-                await message.reply(content=description, mention_author=False)
+        if not (link_type and link_url):
+            return
+
+        if link_type == LinkType.TIKTOK:
+            await self._send_embed(
+                link_url, "TikTok Link", message, MediaLinkFormatter.embed_tiktok
+            )
 
-            elif link_type == "twitter":
-                return await self.mark_twitter_post(message)
+        if link_type == LinkType.TWITTER:
+            return await self.mark_post_emoji(message)
```

Comment:
Maybe move the twitter if statement above tiktok since twitter returns early, looks nicer